core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -2604,18 +2604,19 @@ private[spark] object Utils extends Logging {
    *   - serializer(such as KryoSerializer) supports relocation of serialized objects
    */
   def isPushBasedShuffleEnabled(conf: SparkConf): Boolean = {
+    val shouldDoPushBasedShuffle = conf.get(PUSH_BASED_SHUFFLE_ENABLED)
     val serializer = Utils.classForName(conf.get(SERIALIZER)).getConstructor(classOf[SparkConf])
       .newInstance(conf).asInstanceOf[Serializer]
     val canDoPushBasedShuffle =
-      conf.get(PUSH_BASED_SHUFFLE_ENABLED) &&
+      shouldDoPushBasedShuffle &&
         (conf.get(IS_TESTING).getOrElse(false) ||
           (conf.get(SHUFFLE_SERVICE_ENABLED) &&
             conf.get(SparkLauncher.SPARK_MASTER, null) == "yarn" &&
             // TODO: [SPARK-36744] needs to support IO encryption for push-based shuffle
             !conf.get(IO_ENCRYPTION_ENABLED) &&
             serializer.supportsRelocationOfSerializedObjects))
```

Comment:
Can we move this block if `shouldDoPushBasedShuffle` is true ?
Something like:
```
  val pushBasedShuffleEnabled = conf.get(PUSH_BASED_SHUFFLE_ENABLED)
  if (pushBasedShuffleEnabled) {
    val serializer = ...
    val canDoPushBasedShuffle = conf.get(IS_TESTING).getOrElse(false) ||
      (conf.get(SHUFFLE_SERVICE_ENABLED) && ...

    if (!canDoPushBasedShuffle) {
      logWarning(" ...")
    }

    canDoPushBasedShuffle
  } else {
    false
  }
}

```

Reply:
Fixed.